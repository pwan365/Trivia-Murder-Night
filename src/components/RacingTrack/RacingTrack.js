import { Grid } from "@mui/material";
import React from "react";
import DuckIcon from "../../resources/duck.png";
import GhostIcon from "../../resources/ghost.png";

const RacingTrack = (props) => {
  return (
    <div>
      {Object.entries(props.playerStatus).map(([key, value]) => {
        return (
          <div style={{ display: "flex" }}>
            <p style={{ marginRight: "10px", width: "100px" }}>
              {value.playerName}
            </p>
            {value.alive ? (
              <div
                className="player-board"
                style={{
                  width: "100%",
                  display: "grid",
                  gridTemplateColumns: `repeat(${props.numberOfRounds}, 1fr)`,
                }}
              >
                {props.ghostPos !== 0 && (
                  <Grid
                    className="position-ghost"
                    style={{
                      gridColumnStart: props.ghostPos,
                      display: "flex",
                      justifyContent: "center",
                      alignItems: "center",
                    }}
                  >
                    <img
                      style={{
                        width: "30px",
                        height: "30px",
                        fill: "white",
                      }}
                      src={GhostIcon}
                      alt="ghost"
                    />
                  </Grid>
                )}
                <div
                  className="position-player"
                  style={{
                    gridColumnStart: value.position + 1,
                    display: "flex",
                    justifyContent: "center",
                    alignItems: "center",
                  }}
                >
                  <img
                    style={{
                      width: "30px",
                      height: "30px",
                      transition: "all 2s ease",
                    }}
                    src={DuckIcon}
                    alt="duck"
                  />
                </div>
                <p
                  style={{
                    gridColumnStart: 10,
                    transition: "all 2s ease",
                    display: "flex",
                    justifyContent: "center",
                  }}
                >
                  GOAL
                </p>
              </div>
            ) : (
              <p style={{ color: "red" }}>DIED</p>
            )}
          </div>
        );
      })}
    </div>
  );
};

export default RacingTrack;
