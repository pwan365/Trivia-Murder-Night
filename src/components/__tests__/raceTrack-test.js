import React from "react";
import { create } from "react-test-renderer";
import RacingTrack from "../RacingTrack/RacingTrack";

var playerStatus = {
  id: "4hTFDMJN4tz4So_MAAAL",
  playerName: "player2",
  position: 0,
  roomCode: "K8DYu",
};

const racingTrack = create(
  <RacingTrack playerStatus={playerStatus} ghostPos={0} numberOfRounds={5} />
);

test("raceSnapshot", () => {
  expect(racingTrack).toMatchSnapshot();
});
