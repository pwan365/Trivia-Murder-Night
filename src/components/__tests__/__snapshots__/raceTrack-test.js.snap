// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`raceSnapshot 1`] = `
<div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <p
      style={
        Object {
          "marginRight": "10px",
          "width": "100px",
        }
      }
    />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      DIED
    </p>
  </div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <p
      style={
        Object {
          "marginRight": "10px",
          "width": "100px",
        }
      }
    />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      DIED
    </p>
  </div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <p
      style={
        Object {
          "marginRight": "10px",
          "width": "100px",
        }
      }
    />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      DIED
    </p>
  </div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <p
      style={
        Object {
          "marginRight": "10px",
          "width": "100px",
        }
      }
    />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      DIED
    </p>
  </div>
</div>
`;
